tool/gravity/cli/clusterupdate.go

```diff
@@ -96,6 +107,65 @@ func newClusterUpdater(
 	return updater, nil
 }
 
+// checkStatus returns an error if the cluster is degraded.
+// If force is true, warnings will be ignored.
+func checkStatus(ctx context.Context, force bool) error {
+	nodes, err := statusapi.NodeStatuses(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var failedProbes []string
+	var warningProbes []string
+	for _, node := range nodes {
+		failedProbes = append(failedProbes, node.FailedProbes...)
+		warningProbes = append(warningProbes, node.WarnProbes...)
+	}
+
+	if len(failedProbes) > 0 {
+		return trace.BadParameter("unable to upgrade degraded cluster")
+	}
+
+	if !force && len(warningProbes) > 0 {
+		return trace.BadParameter("cluster has active warnings (use --force to continue)")
```

Comment:
And in case of warning:

```
Some cluster nodes have active warnings:
  // print nodes with warnings
You can provide a --force flag to suppress this message and launch an upgrade anyway.
```